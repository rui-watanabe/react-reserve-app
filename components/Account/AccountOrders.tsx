import React from 'react';
import {
  Header,
  Accordion,
  Label,
  Segment,
  Icon,
  Button,
  List,
  Image,
  Message,
} from 'semantic-ui-react';
import { useRouter } from 'next/router';
import formatDate from '../../utils/formatDate';
import { AccountOrdersProps } from './AccountType';
import { OrderModelType } from '../../models/OrderModel/OrderType';

function AccountOrders({ orders }: AccountOrdersProps): JSX.Element {
  const router = useRouter();

  function mapOrdersToPanels(orders: OrderModelType[]) {
    return orders.map((order) => ({
      key: order._id,
      title: {
        content: <Label color="blue" content={formatDate(order.createdAt)} />,
      },
      content: {
        content: (
          <>
            <List.Header as="h3">
              Total: ${order.total}
              <Label
                content={order.email}
                icon="mail"
                basic
                horizontal
                style={{ marginLeft: '1em' }}
              />
            </List.Header>
            <List>
              {order.products.map((p) => {
                '_id' in p.product ? (
                  <List.Item key={p.product._id}>
                    <Image avatar src={p.product.mediaUrl} />
                    <List.Content>
                      <List.Header>{p.product.name}</List.Header>
                      <List.Description>
                        {p.quantity} ãƒ» ${p.product.price}
                      </List.Description>
                    </List.Content>
                    <List.Content floated="right">
                      <Label tag color="red" size="tiny">
                        {p.product.sku}
                      </Label>
                    </List.Content>
                  </List.Item>
                ) : (
                  <Message negative>
                    <Message.Header>
                      We`re sorry we can`t apply that discount
                    </Message.Header>
                    <p>Setting Product ObjectId error!</p>
                  </Message>
                );
              })}
            </List>
          </>
        ),
      },
    }));
  }

  return (
    <>
      <Header as="h2">
        <Icon name="folder open" />
        Order History
      </Header>
      {orders.length === 0 ? (
        <Segment inverted tertiary color="grey" textAlign="center">
          <Header icon>
            <Icon name="copy outline" />
            No past orders.
          </Header>
          <div>
            <Button onClick={() => router.push('/')} color="orange">
              View Products
            </Button>
          </div>
        </Segment>
      ) : (
        <Accordion
          fluid
          styled
          exclusive={false}
          panels={mapOrdersToPanels(orders)}
        />
      )}
    </>
  );
}

export default AccountOrders;
