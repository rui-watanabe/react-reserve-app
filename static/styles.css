:root {
  --bodyBg: #f0f4f8;
  --menuBg: #008080;
  --progressBar: #54d1db;
}

body {
  background: var(--bodyBg);
}

#menu {
  background: var(--menuBg);
}

/* Overrides default styled accordion styles (too much bottom padding) */
.ui.styled.accordion .content {
  margin: 0;
  padding: 0.5em 1em 0.75em !important;
}

.ui.card > .image:not(.ui) > img, .ui.cards > .card > .image:not(.ui) > img  {
  height: 212px;
}
