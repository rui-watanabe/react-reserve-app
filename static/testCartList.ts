import { CartProductType } from '../src/pagesTypes/CartType';
import { testProductList } from './testProductList';

export const testCartList: CartProductType = [
  {
    quantity: 1,
    product: testProductList[0],
  },
  {
    quantity: 2,
    product: testProductList[1],
  },
];
