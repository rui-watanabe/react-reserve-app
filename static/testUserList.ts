import { ObjectId } from 'mongodb';
import { UserModelType } from '../models/Types/UserType';

export const testUserList: UserModelType[] = [
  {
    _id: new ObjectId('000000000000000000000001'),
    createdAt: new Date('2020-9-22'),
    updatedAt: new Date('2020-9-22'),
    name: 'test',
    email: 'test@gmail.com',
    password: '11111111',
    role: 'root',
  },
  {
    _id: new ObjectId('000000000000000000000002'),
    createdAt: new Date('2020-9-22'),
    updatedAt: new Date('2020-9-22'),
    name: 'test2',
    email: 'test2@gmail.com',
    password: '22222222',
    role: 'admin',
  },
  {
    _id: new ObjectId('000000000000000000000003'),
    createdAt: new Date('2020-9-22'),
    updatedAt: new Date('2020-9-22'),
    name: 'test',
    email: 'test@gmail.com',
    password: '33333333',
    role: 'user',
  },
];
