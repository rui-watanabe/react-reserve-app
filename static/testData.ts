import { ProductModelType } from '../models/Types/ProductType';
import { UserModelType } from '../models/Types/UserType';
import { CartProductType } from '../src/pagesTypes/CartType';
import { testProductList } from './testProductList';
import { testUserList } from './testUserList';

export const testProduct: ProductModelType = testProductList[0];

export const testTotalPages = Math.floor(testProductList.length / 9);

export const testUser: UserModelType = testUserList[2]; // 'user' role user

export const testCartProductList: CartProductType = [
  { product: { ...testProduct }, quantity: 1 },
  { product: { ...testProductList[1] }, quantity: 2 },
  { product: { ...testProductList[2] }, quantity: 3 },
];
