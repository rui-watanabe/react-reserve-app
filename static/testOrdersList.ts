import { ObjectId, ObjectID } from 'mongodb';
import { OrderModelType } from '../models/Types/OrderType';
import { testProductList } from './testProductList';
import { testUserList } from './testUserList';

export const testOrdersList: OrderModelType[] = [
  {
    _id: new ObjectId('200000000000000000000001'),
    createdAt: new Date(2020, 4, 1, 12, 30, 45),
    updatedAt: new Date(2020, 4, 1, 12, 30, 45),
    products: [
      {
        _id: new ObjectID('100000000000000000000001'),
        quantity: 1,
        product: testProductList[0],
      },
      {
        _id: new ObjectID('100000000000000000000002'),
        quantity: 2,
        product: testProductList[1],
      },
    ],
    user: testUserList[2]._id, // 'user' role user
    email: testUserList[2].email,
    total: testProductList[0].price + testProductList[1].price * 2,
  },
  {
    _id: new ObjectID('200000000000000000000002'),
    createdAt: new Date(2020, 4, 1, 12, 30, 45),
    updatedAt: new Date(2020, 4, 1, 12, 30, 45),
    products: [
      {
        _id: new ObjectID('100000000000000000000003'),
        quantity: 1,
        product: testProductList[2],
      },
      {
        _id: new ObjectID('100000000000000000000004'),
        quantity: 2,
        product: testProductList[3],
      },
    ],
    user: testUserList[2]._id, // 'user' role user
    email: testUserList[2].email,
    total: testProductList[2].price + testProductList[3].price * 2,
  },
  {
    _id: new ObjectID('200000000000000000000003'),
    createdAt: new Date(2020, 4, 1, 12, 30, 45),
    updatedAt: new Date(2020, 4, 1, 12, 30, 45),
    products: [
      {
        _id: new ObjectID('100000000000000000000005'),
        quantity: 1,
        product: testProductList[4],
      },
      {
        _id: new ObjectID('100000000000000000000006'),
        quantity: 2,
        product: testProductList[5],
      },
    ],
    user: testUserList[2]._id, // 'user' role user
    email: testUserList[2].email,
    total: testProductList[4].price + testProductList[5].price * 2,
  },
];
