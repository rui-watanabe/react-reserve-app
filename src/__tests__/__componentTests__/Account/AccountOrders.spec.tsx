import React from 'react';
import renderer from 'react-test-renderer';
import { testOrdersList } from '../../../../static/testOrdersList';
import AccountOrders from '../../../components/Account/AccountOrders';

test('AccountOrders', () => {
  const component = renderer.create(<AccountOrders orders={testOrdersList} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
