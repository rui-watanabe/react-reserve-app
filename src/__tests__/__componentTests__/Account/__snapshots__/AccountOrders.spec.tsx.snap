// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountOrders 1`] = `
Array [
  <h2
    className="ui header"
  >
    <i
      aria-hidden="true"
      className="folder open icon"
      onClick={[Function]}
    />
    Order History
  </h2>,
  <div
    className="accordion ui fluid styled"
  >
    <div
      className="title"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="ui blue label"
        onClick={[Function]}
      >
        2020-5-1
      </div>
    </div>
    <div
      className="content"
    >
      <h3
        className="header"
      >
        Total: $
        4598.88
        <div
          className="ui basic horizontal label"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": "1em",
            }
          }
        >
          <i
            aria-hidden="true"
            className="mail icon"
            onClick={[Function]}
          />
          test@gmail.com
        </div>
      </h3>
      <div
        className="ui list"
        role="list"
      />
    </div>
    <div
      className="title"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="ui blue label"
        onClick={[Function]}
      >
        2020-5-1
      </div>
    </div>
    <div
      className="content"
    >
      <h3
        className="header"
      >
        Total: $
        4698.03
        <div
          className="ui basic horizontal label"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": "1em",
            }
          }
        >
          <i
            aria-hidden="true"
            className="mail icon"
            onClick={[Function]}
          />
          test@gmail.com
        </div>
      </h3>
      <div
        className="ui list"
        role="list"
      />
    </div>
    <div
      className="title"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="ui blue label"
        onClick={[Function]}
      >
        2020-5-1
      </div>
    </div>
    <div
      className="content"
    >
      <h3
        className="header"
      >
        Total: $
        4597.74
        <div
          className="ui basic horizontal label"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": "1em",
            }
          }
        >
          <i
            aria-hidden="true"
            className="mail icon"
            onClick={[Function]}
          />
          test@gmail.com
        </div>
      </h3>
      <div
        className="ui list"
        role="list"
      />
    </div>
  </div>,
]
`;
